src/app/layout/qgslayoutlegendwidget.cpp

```diff
@@ -900,6 +903,44 @@ void QgsLayoutLegendWidget::mExpressionFilterButton_toggled( bool checked )
   mLegend->endCommand();
 }
 
+void QgsLayoutLegendWidget::mLayerExpressionButton_clicked()
+{
+
+  if ( !mLegend )
+  {
+    return;
+  }
+
+  QModelIndex currentIndex = mItemTreeView->currentIndex();
+  if ( !currentIndex.isValid() )
+    return;
+
+  QgsLayerTreeNode *currentNode = mItemTreeView->currentNode();
+  if ( !QgsLayerTree::isLayer( currentNode ) )
+    return;
+
+  QgsLayerTreeLayer *LayerNode = qobject_cast<QgsLayerTreeLayer *>( currentNode );
```

Comment:
Variables always start with lowercase.

```suggestion
  QgsLayerTreeLayer *layerNode = qobject_cast<QgsLayerTreeLayer *>( currentNode );
```

Reply:
Done